wagtail/admin/templates/wagtailadmin/shared/breadcrumb-next.html

```diff
@@ -7,51 +7,66 @@
         <div class="w-relative w-h-[50px] w-mr-4 w-top-0 w-z-20 w-flex w-items-center w-flex-row w-flex-1 sm:w-flex-none w-transition w-duration-300">
             <nav class="w-flex w-items-center w-flex-row w-h-full"
                 aria-label="{% trans 'Breadcrumb' %}">
                 <ol class="w-flex w-flex-row w-justify-start w-items-center w-h-full w-pl-0 w-my-0 w-gap-2 sm:w-gap-0 sm:w-space-x-2">
                     {% for page in pages %}
                         {% if page.is_root %}
-                            <li class="{{ breadcrumb_item_classes }} w-max-w-0" data-breadcrumb-item hidden>
-                                <a class="{{ breadcrumb_link_classes }}" href="{% url 'wagtailadmin_explore_root' %}">
+                            <li
+                                class="{{ breadcrumb_item_classes }} {% if not is_expanded %}w-max-w-0{% endif %}"
+                                data-breadcrumb-item
+                                {% if not is_expanded %}hidden{% endif %}
+                            >
+                                <a
+                                    class="{{ breadcrumb_link_classes }}"
+                                    href="{% if url_root_name %}{% url url_root_name %}{% else %}{% url url_name page.id %}{% endif %}{{ querystring_value }}"
+                                >
                                     {% trans "Root" %}
                                 </a>
                                 {% icon name="arrow-right" class_name=icon_classes %}
                             </li>
                         {% elif forloop.first %}
                             {# For limited-permission users whose breadcrumb starts further down from the root #}
-                            <li class="{{ breadcrumb_item_classes }} w-max-w-0" data-breadcrumb-item hidden>
-                                <a class="{{ breadcrumb_link_classes }}" href="{% url 'wagtailadmin_explore' page.id %}">
+                            <li
+                                class="{{ breadcrumb_item_classes }} {% if not is_expanded %}w-max-w-0{% endif %}"
+                                data-breadcrumb-item
+                                {% if not is_expanded %}hidden{% endif %}
+                            >
+                                <a class="{{ breadcrumb_link_classes }}" href="{% url url_name page.id %}{{ querystring_value }}">
                                     {% trans "Root" %}
                                 </a>
                                 {% icon name="arrow-right" class_name=icon_classes %}
                             </li>
                         {% elif forloop.last %}
-                            <li class="{{ breadcrumb_item_classes }}" {% if trailing_breadcrumb_title %} data-breadcrumb-item hidden {% endif %}>
+                            <li class="{{ breadcrumb_item_classes }}" {% if trailing_breadcrumb_title and not is_expanded %} data-breadcrumb-item hidden {% endif %}>
```

Comment:
Just realised that the last item click will not work as expected in the chooser because the jQuery code won't find this one. 
Try to search or select something other than the root folder and pretty sure the link won't work correctly.  

Something like this may be needed. 
```suggestion
                            <li class="{{ breadcrumb_item_classes }}"  {% if trailing_breadcrumb_title or not is_expanded %}data-breadcrumb-item{% end of %} {% if trailing_breadcrumb_title and not is_expanded %}hidden{% endif %}>
```